File: php/features.php
<?php
declare(strict_types=1);

// Realgazeta SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RealgazetaFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RealgazetaBaseFeature();
            case "test":
                return new RealgazetaTestFeature();
            default:
                return new RealgazetaBaseFeature();
        }
    }
}

File: py/realgazeta_sdk/features.py
# Realgazeta SDK feature factory

from realgazeta_sdk.feature.base_feature import RealgazetaBaseFeature
from realgazeta_sdk.feature.test_feature import RealgazetaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RealgazetaBaseFeature(),
        "test": lambda: RealgazetaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
